### PR TITLE
[FIX] fix perf issue with sale_order_lot_selection

### PR DESCRIPTION
**Pull request by sebastienbeau**
Hi, first thank to Agilebg for this module

I have some perf issue with sale_order_selection_lot. Indeed my customer have more than 4000 lot for a product and when he try to select it in a sale order the function spend 3 min to compute the lot available.
With this change it spend less than 0,023s
What do you think?
@hurrinico @eLBati 

Thanks for you're feedback

### CODE REVIEW

File: sale_order_lot_selection/model/sale.py [hunk 1]
```diff
@@ -44,24 +44,19 @@ def product_id_change_with_wh(
...
-                'location': location.id,
-                })._product_available()
-            if qty_res[product]['qty_available'] > 0:
-                if lot.id not in available_lots:
-                    available_lots.append(lot.id)
+        locations = self.env['stock.location'].search([
+            ('id', 'child_of', location.id),
+            ])
+        # Search all lot existing lot for the product and location selected
+        self._cr.execute("""SELECT lot_id
```
Comment (pedrobaeza) [inline sale_order_lot_selection/model/sale.py]: You can use quants for searching, but without SQL. The performance will be for sure still good. You can use read_group function if you want to increase it. This way, you don't overpass multi-company rules, and so on.

Comment (sebastienbeau) [inline sale_order_lot_selection/model/sale.py]: good idea, let's fix it

**Comment by sebastienbeau** [comment]
@pedrobaeza everything is green and use read_group ;)

**Comment by pedrobaeza** [comment]
Thanks :+1:

**Comment by hparfr** [comment]
:+1: 
Less CPU / Memory consumption is better for Earth :evergreen_tree:  :earth_africa:

**Comment by bealdav** [comment]
:+1:

**Comment by eLBati** [comment]
Thanks :smile:

**Comment by hurrinico** [comment]
Really Thank You @sebastienbeau :+1:

**Pull request merged**